pkg/virt-handler/vm.go

```diff
@@ -2425,14 +2425,13 @@ func (d *VirtualMachineController) helperVmShutdown(vmi *v1.VirtualMachineInstan
 		return err
 	}
 
-	// Only attempt to gracefully shutdown if the domain has the ACPI feature enabled
-	if isACPIEnabled(vmi, domain) && tryGracefully {
+	if tryGracefully {
```

Comment:
@vladikr I am not sure here. In the past we would treat grace period missing as `acpi = false` and would kill the domain. Nowadays the grace period should always be there (did not check if that was the case in the past). Missing the metadata means we had to reconstruct the domain our self... In this case I would say we should wait for the client to connect... WDYT?

Reply:
Hi @xpivarc   @vladikr   previous also we are inderectly domainHasGracePeriod()
Hi @xpivarc  @vladikr  
isACPIEnabled() invoking domainHasGracePeriod() it cheking grace period is nil or not  . 

now isACPIEnabled() not valid, so we can change to 
if domainHasGracePeriod(domain) && tryGracefully {
       //old logic
}

so that , my pr not braking  /altering old flow. 

 
 
 code snippet for reference

func vmiHasTerminationGracePeriod(vmi *v1.VirtualMachineInstance) bool {
	// if not set we use the default graceperiod
	return vmi.Spec.TerminationGracePeriodSeconds == nil ||
		(vmi.Spec.TerminationGracePeriodSeconds != nil && *vmi.Spec.TerminationGracePeriodSeconds != 0)
}

func isACPIEnabled(vmi *v1.VirtualMachineInstance, domain *api.Domain) bool {
	return (vmiHasTerminationGracePeriod(vmi) || (vmi.Spec.TerminationGracePeriodSeconds == nil && domainHasGracePeriod(domain))) &&
		domain != nil &&
		domain.Spec.Features != nil &&
		domain.Spec.Features.ACPI != nil
}

Reply:
I am fine with adding `domainHasGracePeriod` check as that would keep the old behavior. Can you create an issue to track the possible wrong decision of killing the domain if the grace period is nil?

Reply:
hi @xpivarc , sorry for late reply , i was on leave
i have created issue [link](https://github.com/kubevirt/kubevirt/issues/13180)

Now all tests are passing also , please could you remove hold and  provide approve/lgtm

Reply:
@xpivarc FYA ^^